hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/io/erasurecode/TestCoderBase.java

```diff
@@ -224,13 +225,25 @@ protected ECChunk[] backupAndEraseChunks(ECChunk[] dataChunks,
     int idx = 0;
 
     for (int i = 0; i < erasedDataIndexes.length; i++) {
-      toEraseChunks[idx ++] = dataChunks[erasedDataIndexes[i]];
-      dataChunks[erasedDataIndexes[i]] = null;
+      if (erasedDataIndexes[i] < dataChunks.length) {
+        toEraseChunks[idx ++] = dataChunks[erasedDataIndexes[i]];
+        dataChunks[erasedDataIndexes[i]] = null;
+      } else {
+        throw new HadoopIllegalArgumentException(
```

Comment:
ditto

Reply:
@steveloughran, sorry I don't quite understand this. Is this also a public API? I found that this is only being used in Test classes. 
I have changed the exception type to IllegalArgumentException.